homeassistant/components/smartthings/config_flow.py

```diff
@@ -240,6 +244,41 @@ def _show_step_pat(self, errors):
             },
         )
 
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle re-authentication of an existing config entry."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
```

Comment:
Any reason for this being it's own step instead of being inside `async_step_reauth` ?

Reply:
Because I kinda feel that we should tell the user what will happen rather than opening and dropping them on a Samsung authentication page. It can give that extra connect where we can explain what happens and what will happen